services/web/server/src/simcore_service_webserver/computation_handlers.py

```diff
@@ -237,45 +237,52 @@ def get_celery(_app: web.Application):
     await _set_tasks_in_tasks_db(db_engine, project_id, tasks)
     log.debug("END OF ROUTINE.")
 
+async def _pre_update_pipeline(request):
+    await check_permission(request, "services.pipeline.*")
+
+    # TODO: PC->SAN why validation is commented???
+    # params, query, body = await extract_and_validate(request)
+    project_id = request.match_info.get("project_id", None)
+    if project_id is None:
+        raise web.HTTPBadRequest
+
+
+    user_id = request[RQT_USERID_KEY]
+
+    project = await get_project_for_user(request, project_id, user_id)
+    pipeline_data = project["workbench"]
+
+    # update pipeline
+    await _update_pipeline_db(request.app, project_id, pipeline_data)
+
+    return user_id, project_id
+
+# HANDLERS ------------------------------------------
+
 @login_required
 async def update_pipeline(request: web.Request) -> web.Response:
-    # retrieve the data
-    project_id = request.match_info.get("project_id", None)
-    assert project_id is not None
-    pipeline_data = (await request.json())["workbench"]
-    app = request.app
-    await _update_pipeline_db(app, project_id, pipeline_data)
-    return web.json_response(status=204)
+    await _pre_update_pipeline(request)
+
+    raise web.HTTPNoContent()
+
 
-# pylint:disable=too-many-branches, too-many-statements
 @login_required
 async def start_pipeline(request: web.Request) -> web.Response:
-    # params, query, body = await extract_and_validate(request)
-    
-    # if params is not None:
-    #     log.debug("params: %s", params)
-    # if query is not None:
-    #     log.debug("query: %s", query)
-    # if body is not None:
-    #     log.debug("body: %s", body)
-
-    # assert "project_id" in params
-    # assert "workbench" in body
-
-    # retrieve the data
-    project_id = request.match_info.get("project_id", None)
-    assert project_id is not None
-    pipeline_data = (await request.json())["workbench"]
-    userid = request.get(RQT_USERID_KEY, ANONYMOUS_USER)
-    app = request.app
-    await _update_pipeline_db(app, project_id, pipeline_data)
+    """ Starts pipeline described in the workbench section of a valid project
+        already at the server side
+    """
+    user_id, project_id = await _pre_update_pipeline(request)
+
     # commit the tasks to celery
-    _ = get_celery(request.app).send_task("comp.task", args=(userid, project_id,), kwargs={})
+    _ = get_celery(request.app).send_task("comp.task", args=(user_id, project_id,), kwargs={})
+
     log.debug("Task commited")
-    # answer the client
+
+    # answer the client while task has been spawned
     data = {
-    "pipeline_name":"request_data",
-    "project_id": project_id
+        # TODO: PC->SAN: some name with task id. e.g. to distinguish two projects with identical pipeline?
```

Comment:
not sure if that is not a left over from before. or maybe to help debugging in celery? maybe ask @mguidon if he remembers... if not you can probably remove it...